bgpd/bgp_mplsvpn.c

```diff
@@ -2691,6 +2699,7 @@ void vpn_leak_no_retain(struct bgp *to_bgp, struct bgp *vpn_from, afi_t afi)
 					continue;
 
 				bgp_unlink_nexthop(bpi);
+				bgp_unlink_te_nexthop(bpi);
 				bgp_rib_remove(bn, bpi, bpi->peer, afi, safi);
```

Comment:
on a long term, I think SRv6 could be applied to not only L3VPN. so it should be better to put bgp_unlink_te_nexthop into bgp_unlink _nexthop.